### DISCUSSION TITLE
macOS document icons proposal

### DISCUSSION THREAD

**Discussion by setanarut** [VSCodium/vscodium#616] [Ideas]
Document icons proposal compatible with the new [macOS Big Sur App icon](https://github.com/VSCodium/vscodium/pull/612)

![slide](https://user-images.githubusercontent.com/36481442/104857312-91866500-5928-11eb-816e-03347bf3f848.png)

Screenshots on Finder with dark and light mode
<img width="1246" alt="Ekran Resmi 2021-01-19 19 59 59" src="https://user-images.githubusercontent.com/36481442/105067913-4c317700-5a91-11eb-8c28-86304276cab0.png">
<img width="1246" alt="Ekran Resmi 2021-01-19 20 01 46" src="https://user-images.githubusercontent.com/36481442/105067933-52bfee80-5a91-11eb-8f44-7b775b28c42b.png">

---

Thanks to @daiyam for related Pull Request https://github.com/VSCodium/vscodium/pull/618

**Comment by setanarut** [comment]
Also, all icons are not properly aligned. there are slight differences. Look carefully at the lower right corner.

![a mov-4569994E-9443-4FAB-8432-4E687EBEADF7](https://user-images.githubusercontent.com/36481442/104858208-bbdb2100-592e-11eb-88fb-f2e66418d931.gif)

**Comment by daiyam** [comment]
@hazarek I've made the PR #618 to generate the icons with a script. I can easily update it with a newer icon if you can provide it.

**Comment by setanarut** [comment]


**Reply by daiyam** [reply]
@hazarek Here what I get
![go](https://user-images.githubusercontent.com/587742/104903037-d1c50200-597f-11eb-9247-194a8f0d7523.png)

**Comment by daiyam** [comment]
All I need is a positive vote to switch to the new icon.

@stripedpajamas What do you think of the new icons?

**Comment by setanarut** [comment]
@daiyam Can you share all generated icns files as zip? I want to examine ..

**Reply by daiyam** [reply]
Here you go: [darwin.zip](https://github.com/VSCodium/vscodium/files/5830669/darwin.zip)

**Reply by daiyam** [reply]
The new icon is overriding at little bit the main icon for `shell`, `powershell` and `xml`. So I will need to move it a bit or resize it. What do you think?

**Comment by setanarut** [comment]
@daiyam smaller fixed version;

![vscvodium_document](https://user-images.githubusercontent.com/36481442/104965326-919d6800-59ef-11eb-97e9-ba5470de0089.png)

**Reply by daiyam** [reply]
[darwin.zip](https://github.com/VSCodium/vscodium/files/5832551/darwin.zip)

**Comment by setanarut** [comment]
@daiyam thanks. Looks good.

**Comment by setanarut** [comment]
This is another matter of discussion I think it should be updated;

![Artboard1](https://user-images.githubusercontent.com/36481442/105071096-332ac500-5a95-11eb-8b11-baa92b307edb.png)

**Comment by stripedpajamas** [comment]
Yeah let's go ahead and use the proposed better contrast icons.

**Comment by sefsh** [comment]
This design doesn't adhere to the [macOS Human Interface Guidelines](https://developer.apple.com/news/?id=5i6jlf4d) in the slightest which will result in a non-native look. The use of small icons at the edge of the document looks more like a visual spillover from old Windows visuals. 

This is what a macOS (Big Sur) document icon looks like:

![](https://devimages-cdn.apple.com/wwdc-services/articles/images/19215656-71F6-410E-A1AD-042664BC6DC7/2048.jpeg)

**Reply by setanarut** [reply]
> This design doesn't adhere to the [macOS Human Interface Guidelines](https://developer.apple.com/news/?id=5i6jlf4d) in the slightest which will result in a non-native look. The use of small icons at the edge of the document looks more like a visual spillover from old Windows visuals.
> 
> This is what a macOS (Big Sur) document icon looks like:
> 
> ![](https://camo.githubusercontent.com/83fa107dab7f00004f71db06d4e544090f76af2af10a79ce00dcaef1cd68330f/68747470733a2f2f646576696d616765732d63646e2e6170706c652e636f6d2f777764632d73657276696365732f61727469636c65732f696d616765732f31393231353635362d373146362d343130452d413141442d3034323636344243364443372f323034382e6a706567)

I know. I deleted my proposal in [this](https://github.com/VSCodium/vscodium/issues/591) issue because it got negative votes.
<img width="759" alt="Ekran Resmi 2021-02-10 22 31 08" src="https://user-images.githubusercontent.com/36481442/107561635-d3d15800-6bef-11eb-9918-7ae9c278022c.png">

**Reply by sefsh** [reply]
Ah, okay. It looks better that way. Typeface seem a bit off but other than that it seem to adhere. I think that the VSCodium brand hints should be scratched too, as it looks a bit messy with all those details while not really communicating the logo with it being that discrete. 

I think that, as shown in the link, macOS HIG prefers the document icons to be more app-agnostic than, say, Windows, meaning that the look of the document icon is detached from its associated application:

![bild](https://user-images.githubusercontent.com/346211/107615061-cd70c980-6c4b-11eb-9f71-03522f090ac6.png)

**Comment by setanarut** [comment]
@yusf There is no point in discussing these. Changes are accepted by voting. people will not accept this proposal.

**Reply by sefsh** [reply]
Which proposal?

**Reply by setanarut** [reply]
> Which proposal?

Your native macOS document icons proposal https://github.com/VSCodium/vscodium/discussions/616#discussioncomment-354463. And my practice;

<img width="373" alt="107561635-d3d15800-6bef-11eb-9918-7ae9c278022c" src="https://user-images.githubusercontent.com/36481442/107855496-9d841a80-6e33-11eb-9fb6-e54cf34c21f0.png">

**Comment by lukateras** [comment]
@hazarek Great work! Could you share a vector version of this higher-contrast icon with white outline, if available? I am thinking of proposing a PR with similar changes to yours for Linux platform.

**Reply by setanarut** [reply]
Which icon? blue macos app icon?

**Reply by setanarut** [reply]
I guess you want the document icon. I have .afdesign files for what you see here. I exported it as pdf and svg. inside the zip file.


[Arşiv.zip](https://github.com/VSCodium/vscodium/files/6063988/Arsiv.zip)

**Comment by Zabriskije** [comment]
Good job @hazarek for pushing the visuals to adhere to Apple HIG.
As I've understood, there's no consent on updating the document icons, right? Such a pity.
I've made my own set in line with Apple HIG [here](https://github.com/Zabriskije/MacOS-Icons/blob/8b17ae1e55126593f0456a1f45b9984394f61409/VSCode/VSCode-Document-Icons.7z) if someone is interested. They're a lot more minimalistic than the current one.
<img width="915" alt="Schermata 2022-04-01 alle 14 50 23" src="https://user-images.githubusercontent.com/101254295/161267216-c6db1b0f-c005-4275-bfa6-e3f452f934d5.png">

**Reply by setanarut** [reply]
@Zabriskije Why CSS is `#`?

**Reply by Zabriskije** [reply]
It was in the default icon for CSS in VSCodium, should it have orange { } instead (to recall the color found in HTML)?

**Reply by setanarut** [reply]
ah, now i understand. a mistake inherited from the original icons. Some icons are not compatible with macOS. For example, I don't know what the `default` icon means. The original VSCode icons have the same issues.

**Reply by Zabriskije** [reply]
Maybe `default` is for when you assign a file to VSCode that is not one of the listed/expected. For example: if I choose a TXT file to be opened with VSCode, the app assign it the `default` icon.
